### PR TITLE
Admin: Show name of user in users table view

### LINKED ISSUE #15230: Show users realname
Grafana user create has the ability to enter realname / displayedname. And a user may modify it's own name - but that is currently not used anywhere.

In Configuration->Users grafana lists login and email (but not users real name)
In Configuration->Teams->Members grafana lists Login/username and email (but not realname)

Suggestion:
In user list add a column for the user's real name and add the posibility to search on name - and do the same for team members

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What this PR does / why we need it**:
Adds the name of the users for UserTable view as requested in issue 15230.

- added Name to OrgUserDTO
- added Name to UserTableView

**Which issue(s) this PR fixes**:
Closes: #15230

**Special notes for your reviewer**:
Ongoing discussion around UX for this change

- does it render well on mobiles?

- the search placeholder indicates; `Filter by name or type`.
it is searching for `user.login` or `user.email` as indicated in: https://github.com/grafana/grafana/blob/3f15170914c3189ee7835f0b19ff500db113af73/public/app/features/users/state/selectors.ts#L5

image
.

### CODE REVIEW

**Review by papagian — CHANGES_REQUESTED** [review]
Thank you for your contribution. It looks good!
As you mentioned, we need to enable also searching by name. So, could you please change the search placeholder to `Filter by email, login or name` and include the name in the filtering condition?
Make sure also to update the [snapshot](public/app/features/users/__snapshots__/UsersActionBar.test.tsx.snap) appropriately so as the tests won't fail.

I would be nice also to do the same for the Team users page as the [issue](https://github.com/grafana/grafana/issues/15230) suggests.
The required modifications are:
- Update [`TeamMemberDTO`](https://github.com/grafana/grafana/blob/3f15170914c3189ee7835f0b19ff500db113af73/pkg/models/team_member.go#L66)
- Modify the [query](https://github.com/grafana/grafana/blob/3f15170914c3189ee7835f0b19ff500db113af73/pkg/services/sqlstore/team.go#L389) for fetching the team members to return also the user name column
- Include name in [TeamMember](https://github.com/grafana/grafana/blob/3f15170914c3189ee7835f0b19ff500db113af73/public/app/types/teams.ts#L12) interface and update the [teamMocks](https://github.com/grafana/grafana/blob/3f15170914c3189ee7835f0b19ff500db113af73/public/app/features/teams/__mocks__/teamMocks.ts)
- [Rename](https://github.com/grafana/grafana/blob/3f15170914c3189ee7835f0b19ff500db113af73/public/app/features/teams/TeamMembers.tsx#L118) `Name` header to `Login`, add a new `Name` header in the Team members table.
- [Add name column](https://github.com/grafana/grafana/blob/3f15170914c3189ee7835f0b19ff500db113af73/public/app/features/teams/TeamMemberRow.tsx#L83) in the team member table and update the [test](https://github.com/grafana/grafana/blob/3f15170914c3189ee7835f0b19ff500db113af73/public/app/features/teams/TeamMemberRow.test.tsx#L72)
- Enable [searching team members](https://github.com/grafana/grafana/blob/3f15170914c3189ee7835f0b19ff500db113af73/public/app/features/teams/state/selectors.ts#L29) using name

**Comment by eleijonmarck** [comment]
> So, could you please change the search placeholder to Filter by email, login or name and include the name in the filtering condition?
If I search for the placeholder, I get it in the following views:

image
```
/home/eleijonmarck/dev/grafana/public/app/core/components/OrgActionBar/OrgActionBar.tsx
/home/eleijonmarck/dev/grafana/public/app/features/datasources/NewDataSourcePage.tsx
/home/eleijonmarck/dev/grafana/public/app/features/users/UsersActionBar.tsx
```

Only changed in:
```
/home/eleijonmarck/dev/grafana/public/app/features/users/UsersActionBar.tsx
```

Thank you so much for the clear instructions on how that would have been enabled to search for the teammember name.

**Comment by eleijonmarck** [comment]
> Make sure also to update the snapshot appropriately so as the tests won't fail.
Could you show me a bit how this would be accomplished or point me to the right place to find information about it.

@papagian

**Comment by papagian** [comment]
@eleijonmarck 
Thank you for making the changes. They look good!
Below you will find the answers to your questions.

> Only changed in:
> 
> ```
> /home/eleijonmarck/dev/grafana/public/app/features/users/UsersActionBar.tsx
> ```

That is correct. The others are for filtering other tables.



> Could you show me a bit how this would be accomplished or point me to the right place to find information about it.

Sorry, I didn't notice that the URL was invalid. The snapshots you need to modify are: 
*   [TeamMemberRow.test.tsx.snap](https://github.com/grafana/grafana/blob/3f15170914c3189ee7835f0b19ff500db113af73/public/app/features/teams/__snapshots__/TeamMemberRow.test.tsx.snap)
* [TeamMembers.test.tsx.snap](https://github.com/grafana/grafana/blob/3f15170914c3189ee7835f0b19ff500db113af73/public/app/features/teams/__snapshots__/TeamMembers.test.tsx.snap)
* [UsersActionBar.test.tsx.snap](https://github.com/grafana/grafana/blob/3f15170914c3189ee7835f0b19ff500db113af73/public/app/features/users/__snapshots__/UsersActionBar.test.tsx.snap)

and after you will be able to run successfully the tests using:
`make test-js
`

**Comment by papagian** [comment]
Great! One last think: you need to sync your branch with grafana master in order the codespell check to run successfully.
thanks!

**Comment by eleijonmarck** [comment]
@papagian 
I get the following error after I ran the following sequence.

```bash
$ git fetch upstream
$ git checkout 15230-show-user-realname
$ git merge upstream/master
```

```bash
$ make test-js
```

```
❯ make test-js
test frontend
yarn test
yarn run v1.19.1
$ grunt test
Version 6.6.0-pre
Running "sasslint:src" (sasslint) task

Running "newer:exec:tslintPackages" (newer) task

Running "exec:tslintPackages" (exec) task
$ tslint -c tslint.json --project tsconfig.json
$ tslint -c tslint.json --project tsconfig.json
$ tslint -c tslint.json --project tsconfig.json
$ tslint -c tslint.json --project tsconfig.json

ERROR: /home/eleijonmarck/dev/grafana/packages/grafana-ui/src/components/Graph/Graph.story.tsx:66:1 - Exceeds maximum line length of 150
ERROR: /home/eleijonmarck/dev/grafana/packages/grafana-ui/src/components/Graph/Graph.story.tsx:77:1 - Exceeds maximum line length of 150
ERROR: /home/eleijonmarck/dev/grafana/packages/grafana-ui/src/components/Modal/Modal.story.tsx:12:1 - Exceeds maximum line length of 150
```

File: pkg/services/sqlstore/team.go [hunk 1]
```diff
@@ -387,8 +387,22 @@ func GetTeamMembers(query *models.GetTeamMembersQuery) error {
...
+		"team_member.team_id",
+		"team_member.user_id",
+		"user.email",
+		"user.name",
+		"user.login",
+		"team_member.external",
+		"team_member.permission",
+		"user_auth.auth_module",
+	)
+	sess.Asc(
```
Comment (papagian) [inline pkg/services/sqlstore/team.go]: I just noticed that you have changed the sorting.
I think it's better to leave it as it was so as it conforms with the [sorting](https://github.com/grafana/grafana/blob/3f7f128a4fe912f3224cce6b507c0ac30a8a9ef3/pkg/services/sqlstore/org_users.go#L132) in the users table and also the users will have the same experience they used to have.

**Comment by papagian** [comment]
Great! I saw that you have figured it by yourself!
Please fix also [this](https://github.com/grafana/grafana/pull/18108/files#r348604429) in order to merge it!

File: pkg/services/sqlstore/team.go (hunk 1 above)
Comment (eleijonmarck) [reply pkg/services/sqlstore/team.go]: I have now set

```golang
	sess.Asc(
		"user.login",
		"user.email",
		"user.name",
	)
```

I am unsure if this was the expected sorting, or if you wanted to have as previously but w. name added later.

```golang
	sess.Asc("user.email", "user.login")
```

Comment (eleijonmarck) [reply pkg/services/sqlstore/team.go]: > I think it's better to leave it as it was so
I see

File: pkg/services/sqlstore/team.go [hunk 2]
```diff
@@ -387,8 +387,18 @@ func GetTeamMembers(query *models.GetTeamMembersQuery) error {
...
+		"team_member.team_id",
+		"team_member.user_id",
+		"user.email",
+		"user.name",
+		"user.login",
+		"team_member.external",
+		"team_member.permission",
+		"user_auth.auth_module",
+	)
+	sess.Asc("user.email", "user.login")
```
Comment (papagian) [inline pkg/services/sqlstore/team.go]: Sorry for not being clear: sort first by login and after by email, as it was before:
`sess.Asc("user.login", "user.email")`

Comment (eleijonmarck) [reply pkg/services/sqlstore/team.go]: @papagian Thank you for all the back and forth and guiding me through the pull request!

:heart:

**Review by papagian — APPROVED** [review]

**Pull request merged**